packages/moonstone/Scrollable/ScrollButtons.js

```diff
@@ -150,26 +150,28 @@ class ScrollButtons extends Component {
 	updateButtons = (bounds) => {
 		const
 			{vertical} = this.props,
+			{nextButtonDisabled, prevButtonDisabled} = this.state,
 			currentPos = vertical ? bounds.scrollTop : bounds.scrollLeft,
 			maxPos = vertical ? bounds.maxTop : bounds.maxLeft,
 			shouldDisablePrevButton = currentPos <= 0,
 			/* If a scroll size or a client size is not integer,
 			   browsers's max scroll position could be smaller than maxPos by 1 pixel.*/
-			shouldDisableNextButton = maxPos - currentPos <= 1;
+			shouldDisableNextButton = maxPos - currentPos <= 1,
+			updatePrevButton = (prevButtonDisabled !== shouldDisablePrevButton),
+			updateNextButton = (nextButtonDisabled !== shouldDisableNextButton);
+		let nextState = null;
+
+		if (updatePrevButton && updateNextButton) {
+			nextState = {prevButtonDisabled: shouldDisablePrevButton, nextButtonDisabled: shouldDisableNextButton};
```

Comment:
If we _let_ _nextState_ a simple empty object _{}_, it looks to possible to reduce _if_ clauses.